packages/inventory-vulnerabilities/src/Components/Filters.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable camelcase */
+
+import { FilterDropdown, SimpleTableFilter } from '@redhat-cloud-services/frontend-components';
+import routerParams from '@redhat-cloud-services/frontend-components-utilities/files/RouterParams';
+import * as _ from 'lodash';
+import debounce from 'lodash/debounce';
+import propTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { filtersCVSSScore, filtersPublishDate, filtersSeverity, filtersShowAll } from '../constants';
+import { fetchSystemCveStatusList } from '../redux/actions';
+
+class Filters extends Component {
+    state = { show_all: 'true', publish_date: 'all', cvss_filter: 'all' };
+
+    difference(object, base) {
+        function changes(object, base) {
+            return _.transform(object, function(result, value, key) {
+                if (!_.isEqual(value, base[key])) {
+                    result[key] = _.isObject(value) && _.isObject(base[key]) ? changes(value, base[key]) : value;
+                }
+            });
+        }
+
+        return changes(object, base);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        if (nextState === this.state) {
+            return false;
+        }
+
+        return true;
+    }
+
+    componentDidMount() {
+        this.props.showStatusList && this.props.fetchStatusList();
+        const urlParams = new URLSearchParams(this.props.location.search);
+        const cvss_filter = urlParams.get('cvss_filter');
+        cvss_filter && this.setState({ ...this.state, cvss_filter }, this.apply);
+        cvss_filter && history.pushState(null, null, window.location.href.split('?')[0]);
+    }
+
+    apply = () => {
+        this.props.apply(this.state);
```

Comment:
This looks really fishy, what you are trying to achieve with this? Sending state to parent component?